### PR TITLE
Allow models to be registered as long as llama model is provided

### PR DESCRIPTION
**Pull request by dineshyv**
This PR allows models to be registered with provider as long as the user specifies a llama model, even though the model does not match our prebuilt provider specific mapping.
Test:
pytest -v -s llama_stack/providers/tests/inference/test_model_registration.py -m "together" --env TOGETHER_API_KEY=<KEY>

### CODE REVIEW

File: llama_stack/providers/utils/inference/model_registry.py [hunk 1]
```diff
@@ -60,8 +60,18 @@ def get_llama_model(self, provider_model_id: str) -> str:
...
 
     async def register_model(self, model: Model) -> Model:
-        model.provider_resource_id = self.get_provider_model_id(
-            model.provider_resource_id
-        )
+        provider_resource_id = self.get_provider_model_id(model.provider_resource_id)
+        if provider_resource_id:
+            model.provider_resource_id = provider_resource_id
+        else:
+            if model.metadata.get("llama_model") is None:
```
Comment (ashwinb) [inline llama_stack/providers/utils/inference/model_registry.py:67]: do we validate that llama-model is a legit llama model?

**Review by ashwinb — COMMENTED** [review]
Where in the documentation should we specify this bit that model registration needs to have a llama-model argument?

File: llama_stack/providers/utils/inference/model_registry.py (hunk 1 above)
Comment (dineshyv) [reply llama_stack/providers/utils/inference/model_registry.py:68]: added a check.

File: llama_stack/providers/tests/inference/test_model_registration.py [hunk 2]
```diff
@@ -17,8 +18,17 @@
...
+        inference_impl, models_impl = inference_stack
+
+        provider = inference_impl.routing_table.get_provider_impl(inference_model)
+        if provider.__provider_spec__.provider_type not in (
+            "meta-reference",
+            "remote::ollama",
+            "remote::vllm",
+            "remote::tgi",
+        ):
+            pytest.skip("70B instruct is too big only for local inference providers")
```
Comment (ashwinb) [inline llama_stack/providers/tests/inference/test_model_registration.py:31]: this error message can be improved a little bit

File: llama_stack/providers/utils/inference/model_registry.py [hunk 3]
```diff
@@ -60,8 +61,33 @@ def get_llama_model(self, provider_model_id: str) -> str:
...
+                    "Please specify a llama_model in metadata or use a supported model identifier"
+                )
+            existing_llama_model = self.get_llama_model(model.provider_resource_id)
+            if existing_llama_model:
+                if existing_llama_model != model.metadata["llama_model"]:
+                    raise ValueError(
+                        f"Provider model id '{model.provider_resource_id}' is already registered to a different llama model: '{existing_llama_model}'"
+                    )
+            else:
+                # Validate that the llama model is a valid CoreModelId
```
Comment (ashwinb) [inline llama_stack/providers/utils/inference/model_registry.py:80]: this is not right or at least not future proof. I think you should only accept huggingface names as values for llama model. so collect a list of all huggingface repos allowed and check against that

**Review by ashwinb — CHANGES_REQUESTED** [review]

File: llama_stack/providers/utils/inference/model_registry.py (hunk 3 above)
Comment (dineshyv) [reply llama_stack/providers/utils/inference/model_registry.py]: done.

**Review by ashwinb — APPROVED** [review]

**Pull request merged**